src/services/logger.service.js

```diff
@@ -48,4 +72,41 @@ function getLogger(req) {
   return logger;
 }
 
-export { walletLoggers, initializeWalletLogger, getLogger };
+/**
+ * @param {HathorWallet} wallet
+ * @param {import('winston').Logger} logger
+ */
+function setupWalletStateLogs(wallet, logger) {
+  const times = {
+    connecting: 0,
+    syncing: 0,
+    processing: 0,
+    ready: 0,
+  };
+  wallet.on('state', state => {
+    switch (state) {
+      case HathorWallet.CONNECTING:
+        times.connecting = Date.now();
+        break;
+      case HathorWallet.SYNCING:
+        times.syncing = Date.now();
+        logger.debug(`state_update[syncing]: time to connect ${(times.syncing - times.connecting) / 1000} seconds`);
+        break;
+      case HathorWallet.PROCESSING:
+        times.processing = Date.now();
+        logger.debug(`state_update[processing]: time to sync ${(times.processing - times.syncing) / 1000} seconds`);
+        break;
+      case HathorWallet.READY:
+        times.ready = Date.now();
+        logger.debug(`state_update[ready]: time to process history ${(times.ready - times.processing) / 1000} seconds`);
+        break;
+      default:
+        break;
+    }
+  });
+  wallet.conn.on('wallet-load-partial-update', data => {
+    logger.debug(`Found ${data.addressesFound} addresses and ${data.historyLength} transactions`);
```

Comment:
question: Do we have any moment when this listener is no longer needed? If so, can we remove it at this time?

Reply:
This event is emitted when the wallet is syncing, since we may sync again if the connection is lost we should leave this on.